packages/Webkul/Shop/src/Http/Controllers/ProductsCategoriesProxyController.php

```diff
@@ -49,6 +50,19 @@ public function index(Request $request)
                 if ($product->visible_individually && $product->url_key) {
                     $customer = auth()->guard('customer')->user();
 
+                    $productArr = json_encode([$product->sku]);
+                    $productsData = Cookie::get('product');
+
+                    if (! isset($productsData)) {
+                        Cookie::queue(Cookie::make('product', $productArr, time() + (86400 * 30), "/"));
+                        
+                    } else {
+                        $productArr = json_decode($productsData);
+                        array_push($productArr, $product->sku);
+                        $productArr = json_encode($productArr);
+                        Cookie::queue(Cookie::make('product', $productArr, time() + (86400 * 30), "/"));
+                    }
```

Comment:
This needs to be improvised as variable namings are not good.

Reply:
fixed